cmd/broker/provisioning_test.go

```diff
@@ -1227,3 +1260,185 @@ func TestProvisioning_PRVersionWithoutOverrides(t *testing.T) {
 	// then
 	suite.WaitForProvisioningState(opID, domain.Failed)
 }
+
+func TestProvisioning_ModulesWithGivenCustomModules(t *testing.T) {
+	// given
+	suite := NewBrokerSuiteTest(t)
+	defer suite.TearDown()
+	iid := uuid.New().String()
+
+	// when
+	resp := suite.CallAPI("PUT", fmt.Sprintf("oauth/v2/service_instances/%s?accepts_incomplete=true", iid),
+		`{
+				"service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+				"plan_id": "4deee563-e5ec-4731-b9b1-53b42d855f0c",
+				"context": {
+						"globalaccount_id": "whitelisted-global-account-id",
+						"subaccount_id": "sub-id",
+						"user_id": "john.smith@email.com"
+				},
+				"parameters": {
+					"name": "test",
+					"region": "eastus",
+					"modules": {
+						"list": [
+							{
+								"name": "keda",
+								"channel": "fast"
+							}
+						]
+					}
+				}
+			}`)
+	opID := suite.DecodeOperationID(resp)
+
+	suite.processProvisioningAndReconcilingByOperationID(opID)
+
+	suite.WaitForOperationState(opID, domain.Succeeded)
+	op, err := suite.db.Operations().GetOperationByID(opID)
+	assert.NoError(t, err)
+	assert.YAMLEq(t, internal.GetFileWithTest(t, fmt.Sprintf("%s/%s/%s", testDataPath, kymaTemplate, "kyma-expected-output-1.yaml")), op.KymaTemplate)
```

Comment:
let's name the yaml file with better names, like "XXX-with-keda.yaml, XXX-no-modules.yaml"